week-1-ts-ecosystem/w01e03-object-oriented-structure/src/types/product.ts

```diff
@@ -0,0 +1,21 @@
+export interface Product<Type extends ProductType> {
+  name: string;
+  id: string;
+  price: number;
+  type: Type;
+  count: number;
+}
+
+export enum ProductType {
+  BuyNow = 'buyNow',
+  Auction = 'auction',
+  Free = 'free'
+}
+
+export interface NewProductArgs<Type extends ProductType> {
+  name: string;
+  price: number;
+  type: Type;
+}
+
+export type UpdateProductArgs<Type extends ProductType> = Partial<NewProductArgs<Type>>;
```

Comment:
To jest ciekawie przygotowane - jednak powinno opiera się o `Product` jako o klasę.
Inteface `Product` jest tutaj nadmiarowy